src/CellData/DomainContributions.jl

```diff
@@ -113,23 +113,77 @@ function get_array(a::DomainContribution)
   a.dict[first(keys(a.dict))]
 end
 
-struct Measure <: GridapType
+abstract type Measure <: GridapType end
+
+function integrate(f,b::Measure)
+  @abstractmethod
+end
+
+function get_cell_quadrature(b::Measure)
+  @abstractmethod
+end
+
+function get_cell_points(a::Measure)
+  quad = get_cell_quadrature(a)
+  return get_cell_points(quad)
+end
+
+function (*)(a::Integrand,b::Measure)
+  integrate(a.object,b)
+end
+
+(*)(b::Measure,a::Integrand) = a*b
+
+struct GenericMeasure <: Measure
   quad::CellQuadrature
 end
 
-get_cell_points(a::Measure) = get_cell_points(a.quad)
+Measure(q::CellQuadrature) = GenericMeasure(q)
 
-Measure(args...;kwargs...) = Measure(CellQuadrature(args...;kwargs...))
+Measure(args...;kwargs...) = GenericMeasure(CellQuadrature(args...;kwargs...))
 
-function integrate(f,b::Measure)
+get_cell_quadrature(a::GenericMeasure) = a.quad
+
+function integrate(f,b::GenericMeasure)
   c = integrate(f,b.quad)
   cont = DomainContribution()
   add_contribution!(cont,b.quad.trian,c)
   cont
 end
 
-function (*)(a::Integrand,b::Measure)
-  integrate(a.object,b)
+"""
+  Composite Measure
+
+  Measure such that the integration and target triangulations are different. 
+
+  - ttrian: Target triangulation, where the domain contribution lives.
+  - itrian: Integration triangulation, where the integration takes place.
+  - quad  : CellQuadrature, defined in itrian
+"""
+struct CompositeMeasure{A<:Triangulation,B<:Triangulation,C<:CellQuadrature} <: Measure
+  ttrian :: A
+  itrian :: B
+  quad   :: C
 end
 
-(*)(b::Measure,a::Integrand) = a*b
+function Measure(ttrian::Triangulation,itrian::Triangulation,q::CellQuadrature)
+  @check get_triangulation(q) === itrian
+  return CompositeMeasure(ttrian,itrian,q)
+end
+
+Measure(ttrian::Triangulation,itrian::Triangulation,args...;kwargs...) = CompositeMeasure(ttrian,itrian,args...;kwargs...)
+
+function CompositeMeasure(ttrian::Triangulation,itrian::Triangulation,args...;kwargs...)
+  quad = CellQuadrature(itrian,args...;kwargs...)
+  return CompositeMeasure(ttrian,itrian,quad)
+end
+
+get_cell_quadrature(a::CompositeMeasure) = a.quad
+
+function CellData.integrate(f,b::CompositeMeasure)
+  ic = CellData.integrate(f,b.quad)
+  cont = DomainContribution()
+  tc = move_contributions(ic,b.itrian,b.ttrian)
+  add_contribution!(cont,b.ttrian,tc)
+  cont
+end
```

Comment:
Do we have tests for CompositeMeasure?

Reply:
We are testing all the constructors and using the integration routines as part of FEFunction projection routines.

Reply:
Ok. I was looking for `CompositeMeasure` calls in the tests, but you are leveraging the `Measure` constructure. All good. Sorry for the missunderstanding.